pontoon/contributors/views.py

```diff
@@ -411,13 +423,122 @@ def settings(request):
+    except KeyError:
+        return JsonResponse(
+            {"status": "error", "message": "Token name is required."}, status=400
+        )
+    if request.method == "POST":
+        pat_count = PersonalAccessToken.objects.filter(user=request.user).count()
+
+        if pat_count < PERSONAL_ACCESS_TOKEN_MAX_COUNT:
+            create_token_form = forms.CreateTokenForm(request.POST, user=request.user)
+
+            if create_token_form.is_valid():
+                token = create_token_form.save(commit=False)
+
+                token_unhashed = generate_unhashed_token()
+                date_midnight = timezone.now().date() + timedelta(days=365)
+                token.name = name
+                token.token_hash = make_password(token_unhashed)
+                token.expires_at = timezone.make_aware(
+                    datetime.datetime.combine(date_midnight, time.min)
+                )
+                token.user = request.user
+                token.save()
+                token_id = token.id
+                token_secret = f"{token_id}_{token_unhashed}"
+
+                return JsonResponse(
+                    {
+                        "status": "success",
+                        "message": "Form submitted successfully!",
+                        "data": {
+                            "new_token_id": token.id,
+                            "new_token_name": token.name,
+                            "new_token_last_used": token.last_used.strftime(
+                                "%B, %d, %Y"
+                            )
+                            if token.last_used
+                            else None,
+                            "new_token_expires_at": token.expires_at.strftime(
+                                "%B, %d, %Y"
+                            ),
+                            "new_token_secret": token_secret,
+                        },
+                    }
+                )
+            else:
+                return JsonResponse(
+                    {
+                        "status": "error",
+                        "message": "Form validation failed.",
+                        "errors": create_token_form.errors,
+                    },
+                    status=400,
+                )
+        else:
+            return JsonResponse(
+                {
+                    "status": "error",
+                    "message": "Maximum number of personal access tokens reached.",
+                    "errors": {
+                        "maximum": [
+                            "You can only have up to 10 personal access tokens."
+                        ]
+                    },
+                },
+                status=400,
+            )
+    return JsonResponse(
+        {"status": "error", "message": "Invalid request method."}, status=405
+    )
+
+
+@login_required(redirect_field_name="", login_url="/403")
+@require_AJAX
+def delete_token(request, token_id):
+    if request.method == "POST":
```

Comment:
Use either `require_POST` or the `DELETE` method and flip the condition to reduce indentation.